Sources/AccountSDKIOSWeb/User/User.swift

```diff
@@ -204,61 +204,71 @@ extension User: Equatable {
 extension User {
     
     /// TokenRefreshRequestHandler is responsible for calling refresh once,  and queuing subsequent requests to wait for the One refresh.
-    class TokenRefreshRequestHandler {
-        
-        var isTokenRefreshing: State = .notRefreshing
-        var stateLock = NSLock()
-        
-        enum State {
+    final class TokenRefreshRequestHandler {
+
+        private enum State {
             case isRefreshing
             case notRefreshing
         }
-        
-        let queue = DispatchQueue(label: "NetworkRefreshRequestHandler", attributes: .concurrent)
-        private var requestsOnRefreshFailure: [(Result<UserTokens, RefreshTokenError>) -> Void] = []
-        private var requestsOnRefreshSuccess: [DispatchWorkItem] = []
-        private var completionsOnRefreshWithoutRetry: [(Result<UserTokens, RefreshTokenError>) -> Void] = []
-        
+
+        private let isTokenRefreshing: Synchronized<State> = .init(.notRefreshing)
+        private let requestsOnRefreshFailure: Synchronized<[(Result<UserTokens, RefreshTokenError>) -> Void]> = .init([])
+        private let requestsOnRefreshSuccess: Synchronized<[DispatchWorkItem]> = .init([])
+        private let completionsOnRefreshWithoutRetry: Synchronized<[(Result<UserTokens, RefreshTokenError>) -> Void]> = .init([])
+        private let tokenRefresher: UserTokensRefreshing
+        private let requestMaker: UserRequestMaking
+
+        init(
+            tokenRefresher: UserTokensRefreshing = DefaultUserTokensRefresher(),
+            requestMaker: UserRequestMaking = DefaultUserRequestMaker()
+        ) {
+            self.tokenRefresher = tokenRefresher
+            self.requestMaker = requestMaker
+        }
```

Comment:
Those 2 types make this class testable for our needs, for now. I didn't want to change external APIs

Reply:
Sure, i understand this concept